python-lib/dku_deeplearning_image/utils.py

```diff
@@ -77,25 +78,39 @@ def can_use_gpu():
     return len(tf.config.experimental.list_physical_devices('GPU')) > 0
 
 
-def set_gpu_options(should_use_gpu, gpu_list, memory_limit):
+def set_gpu_options(should_use_gpu, gpu_list, gpu_memory_allocation_mode, memory_limit_ratio=None):
```

Comment:
Just a nice to have, but type hints would help understand what you expect here

Reply:
True, will also dedicate a PR to typing and documentation